erigon-lib/state/aggregator_v3.go

```diff
@@ -764,11 +764,31 @@ func (ac *AggregatorV3Context) CanUnwindBeforeBlockNum(blockNum uint64, tx kv.Tx
 	return blockNumWithCommitment, true, nil
 }
 
+// PruneSmallBatches is not cancellable, it's over when it's over or failed.
+// It fills whole timeout with pruning by small batches (of 100 keys) and making some progress
+func (ac *AggregatorV3Context) PruneSmallBatches(ctx context.Context, timeout time.Duration, tx kv.RwTx) error {
+	localTimeout, cancel := context.WithTimeout(context.Background(), timeout)
+	defer cancel()
+
+	for {
+		if err := ac.Prune(context.Background(), tx, 100); err != nil {
```

Comment:
Need check that prune with  limit is “consistent” - for example Index.Prune doesn ‘t exit before collector.Load (if limit reached).

Reply:
thanks for pointing on it